Lib/asyncio/unix_events.py

```diff
@@ -757,12 +757,18 @@ def _start(self, args, shell, stdin, stdout, stderr, bufsize, **kwargs):
             # other end).  Notably this is needed on AIX, and works
             # just fine on other platforms.
             stdin, stdin_w = socket.socketpair()
-        self._proc = subprocess.Popen(
-            args, shell=shell, stdin=stdin, stdout=stdout, stderr=stderr,
-            universal_newlines=False, bufsize=bufsize, **kwargs)
-        if stdin_w is not None:
-            stdin.close()
-            self._proc.stdin = open(stdin_w.detach(), 'wb', buffering=bufsize)
+        try:
+            self._proc = subprocess.Popen(
+                args, shell=shell, stdin=stdin, stdout=stdout, stderr=stderr,
+                universal_newlines=False, bufsize=bufsize, **kwargs)
+            if stdin_w is not None:
+                self._proc.stdin = open(stdin_w.detach(), 'wb', buffering=bufsize)
+                stdin.close()
```

Comment:
Please move the line before `open()` call.
It saves 1 open file descriptor.
Not a very big deal but why not?

Reply:
I'm not sure I even understand this code :) What's it for?  Why does monkeypatching `self._proc.stdin` work?

Reply:
@1st1 please read a code comment at the begin of the method.
AIX requires to make stdin a socket pair (I love all Unix flavors and their differences).
`proc.stdin` should be a writable file-like object. 
`open()` creates a file object from write-fd of created socket pair.

It exists in asyncio starting from very old times.

Reply:
@asvetlov: Good point. I suppose since `open(stdin_w.detach(), ...)` cannot fail I should have kept the order.

Reply:
> @1st1 please read a code comment at the begin of the method.

Yeah, that's what I did first.  The comment doesn't explain why we pass the *read* end of the pipe to Popen initially (I see that we then monkey-patch it to the *write* end).

> It exists in asyncio starting from very old times.
 
I've no problem with the code or this PR, but I wish that the comment is a bit more elaborate.

Reply:
@1st1 you know, I'm bad in comment writing, sorry.
Would you propose exact text?